src/package-linker.js

```diff
@@ -483,6 +500,16 @@ export default class PackageLinker {
     return range === '*' || satisfiesWithPreleases(version, range, this.config.looseSemver);
   }
 
+  _isChildOfLinkedDep(linkedDepLocations: Array<string>, location: string): boolean {
+    let isWithinLink = false;
+    linkedDepLocations.forEach(loc => {
+      if (path.join(location, 'node_modules').startsWith(loc)) {
```

Comment:
We actually have a constant for this in constants.js: `NODE_MODULES_FOLDER` can you use that instead?

Reply:
Instead of the "heavy" `path.join` can we not just check for `location.startsWith(loc)`?. Also I think using a short-circuited for-of loop would be better here.